Assets/Src/Scripts/SeedCalc/Calculator.cs

```diff
@@ -0,0 +1,233 @@
+    public const string Equal = "=";
+    public const string N0 = "0";
+    public const string N1 = "1";
+    public const string N2 = "2";
+    public const string N3 = "3";
+    public const string N4 = "4";
+    public const string N5 = "5";
+    public const string N6 = "6";
+    public const string N7 = "7";
+    public const string N8 = "8";
+    public const string N9 = "9";
+  }
+
+  // The calculator class. It is a ViewModel class that manages the underlying calculation states,
+  // e.g., via a SeedLang executor.
+  public class Calculator : ViewModel {
+    private class Visualizer : IVisualizer<BinaryEvent>, IVisualizer<EvalEvent> {
+      // The code range that is under execution.
+      public TextRange SourceRange { get; private set; }
+      // The result of the current execution step.
+      public IValue Result { get; private set; }
+
+      public void On(BinaryEvent be) {
+        if (be.Range is TextRange range) {
+          SourceRange = range;
+        }
+        Result = be.Result;
+      }
+
+      public void On(EvalEvent ee) {
+        if (ee.Range is TextRange range) {
+          SourceRange = range;
+        }
+        Result = ee.Value;
+      }
+    }
+
+    private const string _moduleName = "SeedCalc";
+    private const string _evalPrefix = "eval ";
+
+    [BindableProperty]
+    public CalculatorState State {
+      get => _state;
+      set => MvvmSetter(ref _state, value);
+    }
+
+    [BindableProperty]
+    public DisplayContent DisplayContent {
+      get => _displayContent;
+      set => MvvmSetter(ref _displayContent, value);
+    }
+
+    private CalculatorState _state;
+    private const int _maxChars = 100;
+    private readonly StringBuilder _cachedInput = new StringBuilder();
+    private DisplayContent _displayContent = null;
+
+    public void OnInput(string input) {
+      // A simple state machine.
+      if (State.IsError()) {
+        switch (input) {
+          case CalculatorInput.AllClear:
+            ResetState();
+            AllClear();
+            break;
+          case CalculatorInput.Del:
+            ResetState();
+            Backspace();
+            Parse();
+            break;
+          case CalculatorInput.Equal:
+          default:
+            // Neither a printable input nor the Equal key is accepted if the calculator is in an
+            // error state.
+            break;
+        }
+      } else {
+        switch (input) {
+          case CalculatorInput.AllClear:
+            AllClear();
+            break;
+          case CalculatorInput.Del:
+            Backspace();
+            Parse();
+            break;
+          case CalculatorInput.Equal:
+            Execute();
+            break;
+          case CalculatorInput.Dot:
+            // Adds the leading zero before a single dot.
+            if (_cachedInput.Length <= 0) {
+              Append(CalculatorInput.N0);
+            }
+            Append(CalculatorInput.Dot);
+            Parse();
```

Comment:
Need check if State.IsOk() before Parse(), because Append() method may change the state to overflow?